kafka-rest/src/main/java/io/confluent/kafkarest/entities/EmbeddedFormat.java

```diff
@@ -100,4 +100,16 @@ public SchemaProvider getSchemaProvider() {
   public abstract boolean requiresSchema();
 
   public abstract SchemaProvider getSchemaProvider();
+
+  public static EmbeddedFormat forSchemaType(String schemaType) {
+    if (schemaType.equals(AVRO.getSchemaProvider().schemaType())) {
+      return AVRO;
+    } else if (schemaType.equals(JSONSCHEMA.getSchemaProvider().schemaType())) {
+      return JSONSCHEMA;
+    } else if (schemaType.equals(PROTOBUF.getSchemaProvider().schemaType())) {
+      return PROTOBUF;
+    } else {
+      throw new IllegalArgumentException(String.format("Illegal schema type: %s", schemaType));
+    }
+  }
```

Comment:
We want a `NullPointerException`. `schemaType` is not supposed to be `null`. If it is, we want to know.

Reply:
I think it would be nicer to handle it explicitly or have it fall-through to the `IllegalArgumentException` (which will have message _Illegal schema type: null_), instead of getting a generic `NullPointerException`, but I'm OK with this too.

Reply:
Ack.